tests/Feature/Http/Controllers/PostControllerTest.php

```diff
@@ -2,27 +2,167 @@
+        $this->moderator = User::factory()->create()->assignRole('moderator');
+        $this->category = Category::factory()->create();
+        $this->post = new Post([
+            'title' => 'Test',
+            'content' => 'Test',
+        ]);
+        $this->post->category()->associate($this->category);
+        $this->user->posts()->save($this->post);
+
+        $this->post->category()->associate($this->category);
+        $this->user->posts()->save($this->post);
+
+    }
+
     /**
      * @test
      */
     public function it_should_return_the_post_create_view()
     {
-        $user = User::factory()->create()->assignRole('user');
-        $category = Category::factory()->create();
+        $response = $this->actingAs($this->user)->get(route('posts.create', $this->category));
+        $response->assertViewIs('posts.create');
+    }
 
-        $response = $this->actingAs($user)->get(route('posts.create', $category));
+    /**
+     * @test
+     */
+    public function a_moderator_can_delete_any_posts()
+    {
 
-        $response->assertViewIs('posts.create');
+        $this->assertDatabaseHas('posts', [
+            'title' => 'Test',
+            'content' => 'Test',
+            'category_id' => $this->category->id,
+            'user_id' => $this->user->id,
+            'deleted_at' => null
+        ]);
+        
+        $this->actingAs($this->moderator);
+        $this->delete(route('posts.destroy', $this->post));
+        $this->assertSoftDeleted('posts', [
+            'id' => $this->post->id,
+            'title' => 'Test',
+            'content' => 'Test',
+            'category_id' => $this->category->id,
+            'user_id' => $this->user->id,
+        ]);
+       
+    }
+
+    /**
+     * @test
+     */
+    public function a_user_can_delete_own_posts()
+    {
+
+        $this->assertDatabaseHas('posts', [
+            'title' => 'Test',
+            'content' => 'Test',
+            'category_id' => $this->category->id,
+            'user_id' => $this->user->id,
+            'deleted_at' => null
+        ]);
+        
+        $this->actingAs($this->user);
+        $this->delete(route('posts.destroy', $this->post));
+        $this->assertSoftDeleted('posts', [
+            'id' => $this->post->id,
+            'title' => 'Test',
+            'content' => 'Test',
+            'category_id' => $this->category->id,
+            'user_id' => $this->user->id,
+        ]);
     }
+
+    /**
+     * @test
+     */
+    public function a_user_cannot_delete_posts_from_other_users()
+    {
+
+        $this->assertDatabaseHas('posts', [
+            'title' => 'Test',
+            'content' => 'Test',
+            'category_id' => $this->category->id,
+            'user_id' => $this->user->id,
+            'deleted_at' => null
+        ]);
+        
+        $this->user2 = User::factory()->create()->assignRole('user');
+
+        $this->actingAs($this->user2);
+        $this->delete(route('posts.destroy', $this->post));
+
```

Comment:
At this point it would be nice to check which response the user did get back, something like "assertStatus(403)" or similar. Can we store the result of the delete call in a variable and check this result? Like: `$response = $this->delete(route(...));` and then check the result?